src/menhir.ml

```diff
@@ -0,0 +1,32 @@
+open Import
+
+let gen_rules sctx ~dir ~scope (t : Jbuild.Menhir.t) =
+  let targets n = List.map ~f:(Path.relative dir) [n ^ ".ml"; n ^ ".mli"] in
+  let flags =
+    List.map ~f:(Super_context.expand_vars sctx ~scope ~dir) t.flags in
+  let menhir ~extra_targets =
+    Build.run ~extra_targets
+      (Super_context.resolve_program sctx ~hint:"opam install menhir" "menhir")
```

Comment:
This should be outside of the `let menhir` function, so that it's cached between the calls